### PR TITLE
Node 8 and 10

### PR DESCRIPTION
**Pull request by jonstacks**
The diff is a mess for now as it depends on all of the work done for #157. If that gets merged, I can rebase this onto that to clean up the diff. The last few commits are the important ones.

*Edit*: I have rebased this on what is out there.

### CODE REVIEW

**Comment by jonstacks** [comment]
Not sure why its not showing up on here, but [here](https://travis-ci.org/C2FO/patio/builds/514371775) is the travis-ci tests. Works on node 8 & node 10 as well.

File: lib/database/schemaGenerators.js [hunk 1]
```diff
@@ -467,7 +467,7 @@ var AlterTableGenerator = define(null, {
          */
         addForeignKey:function (name, table, opts) {
             opts = opts;
-            if (isArray(name)) {
+            if (Array.isArray(name)) {
```
Comment (dustinsmith1024) [inline lib/database/schemaGenerators.js]: i think there are more of these in this file

Comment (dustinsmith1024) [reply lib/database/schemaGenerators.js]: if not you can remove the isArray var setting at the top of the file

Comment (jonstacks) [reply lib/database/schemaGenerators.js]: I'll go ahead and move them over for consistency. I'm guessing maybe the tests don't hit them or there aren't any issues with them.

Comment (jonstacks) [reply lib/database/schemaGenerators.js]: Fixed in a954338

**Pull request merged**